### PR TITLE
Enable caching using memcached

### PR DESCRIPTION
**Pull request by ASBishop**
Jira: [OSP-26501](https://issues.redhat.com//browse/OSP-26501)

### CODE REVIEW

**Comment by ASBishop** [comment]
/hold

I need to verify something

File: controllers/cinder_controller.go [hunk 1]
```diff
@@ -780,6 +851,7 @@ func (r *CinderReconciler) generateServiceConfigs(
 	instance *cinderv1beta1.Cinder,
 	envVars *map[string]env.Setter,
 	serviceLabels map[string]string,
+	mc *memcachedv1.Memcached,
```
Comment (Akrog) [inline controllers/cinder_controller.go:854]: nit: Wouldn't it make it more readable using `memcached` instead of `mc`?

**Review by Akrog — CHANGES_REQUESTED** [review]
The code looks good, but it is making the deployment of memcached a hard requirement.
I would prefer an approach that would make it optional, for example when the `memcachedInstance` is set to an empty string the `enabled` in the global-defaults is set to `false`.

**Review by ASBishop — COMMENTED** [review]
While I don't know exactly where the discussion took place, my understanding is memcached is a required component (just as it is in tripleo). I recall seeing (but cannot find) a comment in another operator's PR that it is not necessary, at least at this juncture, to provide support for conditionally configuring memcached. See also https://issues.redhat.com/browse/OSP-26273

**Comment by ASBishop** [comment]
/retest

'[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1129)'

**Review by Akrog — APPROVED** [review]
/lgtm

**Pull request merged**